File: src/main/java/fr/inra/oresing/model/BinaryFile.java
package fr.inra.oresing.model;

import fr.inra.oresing.persistence.BinaryFileInfos;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.util.UUID;

@Getter
@Setter
@ToString(callSuper = true)
public class BinaryFile extends OreSiEntity {
    public final static BinaryFileInfos EMPTY_INSTANCE(){
        return new BinaryFileInfos();
    }
    private UUID application;
    private String name;
    private String comment;
    private long size;
    private byte[] data;
    private BinaryFileInfos params;
}

File: ui2/src/model/file/BinaryFile.js
import { BinaryFileInfos } from "./BinaryFileInfos";
export class BinaryFile {
  id;
  params;
  application;
  name;
  size;
  data;
  constructor(params) {
    if (typeof params == "object") {
      Object.keys(this).forEach((key) => (this[key] = params[key] ? params[key] : null));
    }
    this.params = new BinaryFileInfos(this.params);
  }
  get createuser() {
    return (
      this.params &&
      this.params.createuser &&
      this.params.createuser.length &&
      this.params.createuser.slice(0, 8)
    );
  }
  get publisheduser() {
    return (
      this.params &&
      this.params.publisheduser &&
      this.params.publisheduser.length &&
      this.params.publisheduser.slice(0, 8)
    );
  }
}
